projects/objects/buildings/protos/RandomBuilding.proto

```diff
@@ -30,6 +30,7 @@ PROTO RandomBuilding [
   %{
     local wbcore = require('wbcore')
     local wbrandom = require('wbrandom')
+    wbrandom.seed(context.id)
```

Comment:
> This will break existing worlds which rely on the fact that the same building is created every time the world is loaded.

👍🏼 

> We should add the static tag here as well.

I am not sure to understand why adding the static tag is good? IMO it does introduce a bug.

Reply:
Yes, you are right, we should remove it...

Reply:
Either I merged before the comment was made or I missed it. I'll fix it.
Either way I don't think I understand why adding the `static` would create a bug, it's the same logic used in the unit test: [here](https://github.com/cyberbotics/webots/blob/c5cc5452a0c69116a2679ac9de1839909ff95b7b/tests/protos/protos/TemplateStatic.proto), since no seed is provided, it will generate always the same sequence of buildings therefore it's deterministic. Or am I missing something?

For context, [this](https://github.com/cyberbotics/webots/issues/2852) is what triggered the renaming. Specifically, the fact that if no tags are provided, the proto is [regenerated](https://github.com/ad-daniel/webots/blob/4de71628cd276d099cfa0b37b2fb9f9f662d4664/src/webots/vrml/WbNode.cpp#L1834-L1838) . The plan is to make what is currently considered as `static` the default (so if no tags are provided, it will be assumed to be static), and have a tag `nonDeterministic` that handles the cases where the PROTO relies on randomness (time-based seed or user-provided seed).

Reply:
Don't worry, the comment arrived after you merged the PR.
Adding static may create a bug because the proto depends on the node id. If the same proto is instantiated twice in the same world, since it is marked static, it won't be re-evaluated on the second instance and hence will be identical the first one, which is not correct as it has a different node id...

Reply:
Ok, that's what I mean, the node-id based seed is something I added. After [removing](https://github.com/cyberbotics/webots/pull/2893/commits/1d2c45676482e0d486cd7b8db7bb744d1eb955e3) it as per the review, then it is correct that the PROTO can safely be made `static` (in which case, no need to reopen the PR). I was under the impression that the discussion was that it shouldn't be `static` regardless even without a node-id based seed, maybe I misunderstood.

Reply:
I first thought that @DavidMansolino spotted a problem in this PR, but now I believe there is no problem and we should keep the `static` tag here. It is correct to have the `static` tag here as we actually removed the use of the node id. Therefore if a proto is instantiated twice in a world file, it should always produce the same result, so we should keep the static tag here. So, in my opinion, there is nothing to change here.

Reply:
> Don't worry, the comment arrived after you merged the PR.

Yes indeed, sorry for this, as I am reading notifications only from time to time, I hadn't time to comment before the merge.
The problem I see is about the addition of the 'static' tag.
This tag says that if you have two instances of the PROTO node in your world with the same fields values, they will looks exactly the same (and not that they will looks exactly the same each time you load the world), which is not what is wanted here. If you have two random buildings in your world, you want each of them to be different (or not if you are not lucky), which is actually not the case with the 'static' tag (as the fields are the same the procedural part of the PROTO is not re-evaluated).

Reply:
> Therefore if a proto is instantiated twice in a world file, it should always produce the same result because the same seed will be used, hence the same random numbers

Good point, but then, this was a bug, because it was originally created to produce different buildings (and was working at some point, but long time ago, it was probably broken since then).

Reply:
I see. In this case, the original version proposal by @ad-daniel was fixing this bug. We should therefore re-add the `wbrandom.seed(context.id)` and remove the `static` tag to fix the bug...